crates/wasmtime/src/runtime.rs

```diff
@@ -628,6 +629,19 @@ impl Config {
             self.tunables.clone(),
         )
     }
+
+    /// Hashes/fingerprints compiler setting to ensure that compatible
+    /// compilation artifacts are used.
+    pub(crate) fn compiler_fingerprint<H>(&self, state: &mut H)
```

Comment:
Could this include `env!("CARGO_PKG_VERSION")` as well to catch accidental bugs of reusing across wasmtime versions?